src/driver/vectorize.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright DataStax, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Schema, Document, SchemaTypeOptions } from 'mongoose';
+import { StargateMongooseError } from '../stargateMongooseError';
+
+export interface VectorizeOptions extends SchemaTypeOptions<number[]> {
+  service: {
+    provider: string;
+    modelName: string;
+    authentication?: Record<string, unknown>;
+    parameters?: Record<string, unknown>;
+  },
```

Comment:
```suggestion
  service: VectorizeServiceOptions,
```
Just need to add the `VectorizeServiceOptions` from `astra-db-ts`. 

At least you get some extra documentation for free this way.